pandas/core/frame.py

```diff
@@ -4423,6 +4423,8 @@ def insert(
         0   NaN   100     1      99     3
         1   5.0   100     2      99     4
         """
+        if allow_duplicates is lib.no_default:
```

Comment:
no this doesn't make any sense it *must* be the option

Reply:
I don't know what you mean by "the option". There is no "option".
There is a flag, which only partially works, and which is for a different purpose.
This method is DataFrame.insert, which is nothing to do with this PR and which I have no intention of changing.
Fixing the flag is a huge problem and I don't see why you are attaching it to this PR.

Reply:
because you are expanding the api. we cannot expand this api w/o honoring the flag. if that's impossible to do then we cannot proceed. e.g. if i accept this PR then we have a broken option. The point is these are tied together, a partial fix doesn't help at all here.